src/components/layer-panel/index.tsx

```diff
@@ -1,21 +1,35 @@
+import type { RefObject } from 'react';
 import React, { useCallback, useEffect, useState } from 'react';
-import { Card, Tree } from '@douyinfe/semi-ui';
+import { Card, Dropdown, Tree } from '@douyinfe/semi-ui';
 import { observer } from 'mobx-react';
 import { useEditorStore } from '../../store/element';
 import type { ViewElement } from '../../editor/marks/base';
-import type { TreeNodeData } from '@douyinfe/semi-ui/lib/es/tree';
+import type { OnDragProps, RenderFullLabelProps, TreeNodeData } from '@douyinfe/semi-ui/lib/es/tree';
 
-export const LayerPanel = observer(() => {
+import './index.less';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrashCan } from '@fortawesome/free-regular-svg-icons';
+import type { Editor } from '../../editor/editor';
+
+interface Props {
+  editorRef: RefObject<Editor>;
+}
+
+export const LayerPanel = observer(({ editorRef }: Props) => {
   const editorStore = useEditorStore();
 
   const elements = editorStore.viewElements;
   const [treeData, setTreeData] = useState<TreeNodeData[]>([]);
+  const [selectedThroughParent, setSelectedThroughParent] = useState(new Set());
```

Comment:
selectedThroughParent is not need to be operated, why use set here instead of array?

Reply:
There are frequent search operations in node rendering(renderLabel function), hash is more friendly in seaching.